From 10bf1dabf24001629a3f3c552b2002c69e268482 Mon Sep 17 00:00:00 2001
From: tatou-tatou <tatou-tatou@users.noreply.github.com>
Date: Tue, 17 Jun 2014 21:11:34 +0200
Subject: [PATCH] Non-zero return code if nothing is launched

Allow to do the following:
mygtkmenu /path/to/menu || echo "Nothing was launched"
---
 main.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/main.c b/main.c
index 91dcc96..065af53 100644
--- a/main.c
+++ b/main.c
@@ -27,6 +27,7 @@
 
 
 #include <stdio.h>
+#include <stdbool.h>
 #include <gtk/gtk.h>
 #include <string.h>
 #include <unistd.h>
@@ -70,6 +71,7 @@ char *HelpMsg = "myGtkMenu version 1.3, Copyright (C) 2004"
 	"Usage: myGtkMenu MenuDescriptionFilename\n"
 	"TestMenu.txt is an example input file.\n"
 	"See TestMenu.txt for help.\n\n";
+bool exitRunning = false;
 
 
 // ----------------------------------------------------------------------
@@ -97,7 +99,7 @@ int main (int argc, char *argv[]) {
     // If this check causes you problems, comment it out.
     int i_all_ready_running = all_ready_running();
 	if (i_all_ready_running == 1) {
-        fprintf(stderr, "All ready running, will quit.\n");
+        fprintf(stderr, "Already running, will quit.\n");
         return EXIT_FAILURE;
     }
 	else if (i_all_ready_running == 2) {
@@ -336,6 +338,11 @@ int main (int argc, char *argv[]) {
 
 	gtk_main ();
 
+    if (!exitRunning) 
+    {
+        return EXIT_FAILURE;
+    }
+
 	return 0;
 	
 }	// int main
@@ -390,6 +397,9 @@ static void RunItem (char *Cmd) {
 		g_print ("%s\n", error->message);
 		g_error_free (error);
 	}
+    else {
+        exitRunning = true;
+    }
 	gtk_main_quit ();
 }	// static void RunItem
 
-- 
2.0.0

